### PR TITLE
Added generic waitFor and switched existing ones to use this one

### PR DESCRIPTION
**Pull request by konqi**
In my tests i wanted to supply a custom condition for waitFor.
I modified the waitFor so it can handle all kinds of conditions and moved the specifics for query selectors to the respective waitFor functions.

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@jdanyow Can you review this?

**Comment by AdamWillden** [comment]
I'm all for this PR although suspect it may be a breaking change?

Docs will need updating too: 
- http://aurelia.io/hub.html#/doc/article/aurelia/testing/latest/testing-components/8 I've linked to 8 (Helpful Properties and Functions) because link is broken for 9 (Testing complex components)
- https://github.com/aurelia/testing/blob/master/doc/article/en-US/testing-components.md#testing-complex-components

**Comment by konqi** [comment]
I think the docs should be fine - if anything you can do more now.
This should also not be a breaking change - it's hard to tell though with so few tests, which is the reason I added some more tests as well.

File: src/wait.js [hunk 1]
```diff
@@ -1,50 +1,38 @@
...
-    ),
-    wait()
+        rj('condition not met within ' + options.timeout + ' ms');
+      }, options.timeout);
+    }),
+    new Promise((rs) => {
+      const wait = () => {
+        try {
+          let result = condition();
+          if (!options.present ^ !!result) {
```
Comment (AdamWillden) [inline src/wait.js:28]: I find this is a bit confusing... we should use `!options.present === !result` if we want to compare by coercing truthy/falsey values to true or false

Comment (konqi) [reply src/wait.js:28]: I read this `!options.present ^ !!result` as 'either !present or result'. I don't understand what `!options.present === !result` should mean. Those things shouldn't be compared, imho. De Morgan's laws possibly allow it to be refactored like this but I guess a line of comment could help more than applying those.

**Comment by EisenbergEffect** [comment]
Going to close this PR for now, since we've had some pretty big changes in the core library. If this feature remains still interesting to anyone, please submit a new PR based on master. Also, please try to avoid breaking changes if possible.

**Pull request closed without merging**